presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -322,15 +323,18 @@ private void writeStripe(FlushReason flushReason)
 
         columnWriters.forEach(ColumnWriter::close);
 
+        List<DataOutput> outputData = new ArrayList<>();
+        List<Stream> allStreams = new ArrayList<>(columnWriters.size() * 2);
```

Comment:
maybe initializing it with `columnWriters.size() * 3` given each column has one index stream, one data stream, and one present stream? Same for `outputData`.